sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -82,12 +109,25 @@ public struct Behaviors
             _commitBlockHandler += CommitBlockEvent;
         }
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            CleanUp();
+            // We no longer have to read from the channel. We are not expecting any more requests.
+            _stageChunkChannel.Writer.Complete();
+            await _stageChunkChannel.Reader.Completion.ConfigureAwait(false);
+            if (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                _cancellationTokenSource.Cancel();
+            }
+            _cancellationTokenSource.Dispose();
+
+            if (_currentBytesSemaphore != default)
+            {
+                _currentBytesSemaphore.Dispose();
+            }
+            DipsoseHandlers();
```

Comment:
Will cherry pick these changes out and put in a different PR for future review and testing.

https://github.com/Azure/azure-sdk-for-net/issues/32854